src/models/github-server-app.ts

```diff
@@ -163,6 +163,53 @@ export class GitHubServerApp extends EncryptedModel {
 		});
 	}
 
+	/**
+	 * Update existing GitHubServerApp object
+	 *
+	 * @param {{
+	 * 		gitHubClientId: string,
+	 * 		uuid: string,
+	 * 		appId: number;
+	 * 		gitHubBaseUrl: string,
+	 * 		gitHubClientSecret: string,
+	 * 		webhookSecret: string,
+	 * 		privateKey: string,
+	 * 		gitHubAppName: string,
+	 * 		installationId: number
+	 * 	}} payload
+	 */
+
+	static async updateGitHubApp(payload: GitHubServerAppPayload): Promise<void> {
+		const {
+			uuid,
+			appId,
+			gitHubAppName,
+			gitHubBaseUrl,
+			gitHubClientId,
+			gitHubClientSecret,
+			webhookSecret,
+			privateKey,
+			installationId
+		} = payload;
+
+		await this.update(
```

Comment:
Please add some tests.
It'd be great if we can test happy path and some edge cases, like the payload is only partial, will it override all data in config? Or record not found, what's the expectation?

Reply:
Ignore this if you think the tests are cover in another places. :)

Reply:
It would never just be partial. When we direct a user to the editing form, we are auto-populating all the fields with the data from the fb (masking secret stuff). If they try to submit the form with empty fields the frontend validation fails.